### DISCUSSION TITLE
Pagination on v5.0..0

### DISCUSSION THREAD

**Discussion by sidneyrmedeiros** [freekmurze/laravel-analytics#485] [General]
How do I paginate the result in v5.0.0?
I saw that to paginate I need to inform the limit and the offset. But I can only inform the limit.

[Pagination on Google Analytics](https://developers.google.com/analytics/devguides/reporting/data/v1/basics#pagination)

AnalyticsClient.php

```php
public function get(
        string $propertyId,
        Period $period,
        array $metrics,
        array $dimensions = [],
        int $maxResults = 10,
        array $orderBy = [],
    ): Collection {
        $typeCaster = resolve(TypeCaster::class);

        $response = $this->runReport([
            'property' => "properties/{$propertyId}",
            'dateRanges' => [
                $period->toDateRange(),
            ],
            'metrics' => $this->getFormattedMetrics($metrics),
            'dimensions' => $this->getFormattedDimensions($dimensions),
            'limit' => $maxResults,
            'orderBys' => $orderBy,
        ]);
#
#
#
}
    
    ```

**Comment by freekmurze** [comment]
@timvandijck could you take a look at this one?

**Comment by timvandijck** [comment]
We'll have to extend the get method so it will accept and pass the offset. I'll have a look at it, it shouldn't be too hard.

**Comment by timvandijck** [comment]
@sidneyrmedeiros FYI: I just added this to the package: https://github.com/spatie/laravel-analytics/releases/tag/5.0.1

**Reply by sidneyrmedeiros** [reply]
@timvandijck  Thanks for the quick fix! And documentation update! Great job!

**Comment by laudefe** [comment]
Hello @timvandijck ! 
is possible to also add AnalyticsClient.php in the function "get" the parameter "metricFilter" ? 
public function get(
        string $propertyId,
        Period $period,
        array $metrics,
        array $dimensions = [],
        int $maxResults = 10,
        array $orderBy = [],
        int $offset = 0,
        ?FilterExpression $dimensionFilter = null,
        **?FilterExpression $metricFilter= null,**
        bool $keepEmptyRows = false,
    ): Collection

[Reference on Google Analytics](https://cloud.google.com/php/docs/reference/analytics-data/latest/V1beta.BetaAnalyticsDataClient#_Google_Analytics_Data_V1beta_BetaAnalyticsDataClient__runReport__)

**Reply by timvandijck** [reply]
Sounds like a good idea, I don't currently have time to work on it though. I would be happy to accept a PR for this.

**Reply by laudefe** [reply]
Hi! i made the PR [here](https://github.com/spatie/laravel-analytics/pull/523)
